.pipelines/azure_pipeline_validation_appmonitoring.yaml

```diff
@@ -0,0 +1,245 @@
+        pwd
+        az account set --subscription 66010356-d8a5-42d3-8593-6aaa3aeb1c11
+        az aks get-credentials --resource-group appmonitoring-pipeline-test --name aks-pipeline-testbed-ai
+        kubectl delete -f appmonitoring-deployment.yaml
+
+        echo "Wait for 10s for everything to clear up..."
+        sleep 10
+
+        export AIDEV_IMAGE_NAME=${{ variables.fullImageName }}
+        cat appmonitoring-deployment.yaml | envsubst | kubectl apply -f -
+
+        echo "Wait for 10s for charts to be applied..."
+        sleep 10
+
+  - task: AzureCLI@2
+    displayName: "Validate charts deployed to AKS"
+    inputs:
+      azureSubscription: ${{ variables.armServiceConnectionName }}
+      scriptType: bash
+      scriptLocation: inlineScript
+      inlineScript: |
+        cd appmonitoring/scripts
+        pwd
+        az account set --subscription 66010356-d8a5-42d3-8593-6aaa3aeb1c11
+        az aks get-credentials --resource-group appmonitoring-pipeline-test --name aks-pipeline-testbed-ai
+
+        sudo chmod u+x ./validate-aks-charts.sh
+        sudo chmod u+x ./validate-certs.sh
+
+        echo "Checking for charts..."
+        ./validate-aks-charts.sh
+        
+        echo "Checking for certificate installation..."
+        ./validate-certs.sh
+
+  # - task: AzureResourceManagerTemplateDeployment@3
+  #   inputs:
+  #     deploymentScope: 'Resource Group'
+  #     azureResourceManagerConnection: 'ai-acr-validation-rsx'
+  #     subscriptionId: '66010356-d8a5-42d3-8593-6aaa3aeb1c11'
+  #     action: 'Create Or Update Resource Group'
+  #     resourceGroupName: 'appmonitoring-pipeline-test'
+  #     location: 'East US'
+  #     templateLocation: 'Linked artifact'
+  #     csmFile: '/path/to/here'
+  #     deploymentMode: 'Incremental'
+
+  - task: AzureCLI@2
+    displayName: "Install CR and test apps"
+    inputs:
+      azureSubscription: ${{ variables.armServiceConnectionName }}
+      scriptType: bash
+      scriptLocation: inlineScript
+      inlineScript: |
+        cd appmonitoring/helm
+        pwd
+        az account set --subscription 66010356-d8a5-42d3-8593-6aaa3aeb1c11
+        az aks get-credentials --resource-group appmonitoring-pipeline-test --name aks-pipeline-testbed-ai
+        kubectl delete -f java-test-app.yaml
+        kubectl delete -f dotnet-test-app.yaml
+        kubectl delete -f node-test-app.yaml
+        kubectl delete -f appmonitoring-cr.yaml
+
+        echo "Wait for 10s for everything to clear up..."
+        sleep 10
+
+        export JAVA_TEST_IMAGE_NAME=${{ variables.javaTestAppImageName }}
+        export DOTNET_TEST_IMAGE_NAME=${{ variables.dotnetTestAppImageName }}
+        export NODEJS_TEST_IMAGE_NAME=${{ variables.nodeTestAppImageName }}
+        export TEST_NS="${{ variables.testNamespace }}"
+        export AI_CONNECTION_STRING="${{ variables.aiConnectionString }}"
+
+        kubectl create namespace $TEST_NS
+
+        cat appmonitoring-cr.yaml | envsubst | kubectl apply -f -
```

Comment:
Are we seeing kubectl output in the pipeline logs? Will we able to troubleshoot if something fails? Are we failing the entire pipeline here if something fails? We should.

Reply:
these are fixed resources which should not ever fail. We should fail the entire thing if anything fails here as whatever follow is hence rendered moot

Reply:
So is it failing the entire pipeline if it fails? And if it does, do we see output from kubectl to understand why it failed?

Reply:
yes you will fail the whole thing, and it will show the reason as well, just stdout